gengokit/httptransport/httptransport.go

```diff
@@ -276,7 +276,7 @@ func createDecodeConvertFunc(f Field) (string, bool) {
 		needsErrorCheck = false
 	}
 
-	if f.IsEnum {
+	if f.IsEnum && !f.Repeated {
```

Comment:
Consider adding a comment when you run across something like this that might be unexpected.